pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -256,7 +257,9 @@ func (jm *ControllerV2) getJobsToBeReconciled(cronJob *batchv1.CronJob) ([]*batc
 
 	for _, job := range jobList {
 		// If it has a ControllerRef, that's all that matters.
-		if controllerRef := metav1.GetControllerOf(job); controllerRef != nil && controllerRef.Name == cronJob.Name {
+		controllerRef := metav1.GetControllerOf(job)
+		// call resolveControllerRef to make sure the controllerRef is pointing to the right controller
+		if controllerRef != nil && jm.resolveControllerRef(job.Namespace, controllerRef) != nil {
```

Comment:
I think this still appliese: https://github.com/kubernetes/kubernetes/pull/120184#discussion_r1352558833